qutip/solver/integrator/explicit_rk.pyx

```diff
@@ -437,7 +436,7 @@ cdef class Explicit_RungeKutta:
     @cython.cdivision(True)
     cdef double _get_timestep(self, double t):
         """ Get the dt for the step. """
-        cdef double dt_needed = t - self._t_front
+        cdef double dt_needed = t - self._t_prev
```

Comment:
Was this a bug?

Reply:
Yes...
It work fine when for step under tolerance, both values are the same. But when the step was refused, it would return a `dt` a lot smaller than needed for the next step, often dt_needed would be 0, resulting in a step of `min_step`, default 1e-15.

It did not affect output (within tol), but wasted time doing absurdly small steps.